Dodge.js

```diff
@@ -0,0 +1,186 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import AxisPad from 'react-native-axis-pad';
+
+//https://reactnative.dev/docs/timers.html
+//this.setTimeout??
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let index = 0
+
+function Timer(cb, i, seconds) {
+  setTimeout(() => {
+      console.log("Timer is called!")
+      console.log(i)
+      cb(i)
+  }, seconds * 1000);
+}
+
+function DeleteEntity(i){
+delete(ents[i])
+}
+
+function Explode(i){
+  //spawns shrapnel coming from location
+  InitialShrapnel(ents[i].core.x,ents[i].core.y)
+  //delete the original projectile
+  DeleteEntity(i)
+}
+
+function Gameloop(entities, { touches }, )  {
+  //setTimeout is NOT what I want?
+  for (let id in entities) {
+    entities[id].core.update()
+  }
+  
+return entities
+}
+
+function CreateBrick(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateCircle(props){
+  return (
+
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateGrenade(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+  
+
+var ents = {knight: { core: new Knight(), renderer: <CreateBrick />},};
+  //Knight and Brick run every time the game updates that is not good
+
+  function InitialCircle() {
```

Comment:
It might be more general to split these function into two parts:
1) A re-usable "addEntity(entity, renderer)" function, that updates ents[]
2) The "Circle" parts could all be moved into the class Circle.

A few more details:
* The function "InitShrapnel" *should* still exist (somewhere) since it contains logic that relates multiple entities.
* The timer for explode can be moved inside the constructor for Grenade (conceptually, exploding is a *property of* grenades, not of the process that created the Grenade.
* The random positioning could go either way. Either something like:
```
addEntity(new Circle(getRandomX(), getRandomY(), <CreateCircle />)
```
Or maybe the constructor for Circle handles the randomness:
```
class Circle {
  // If no x or y is given, choose one at random.
  constructor(x = getRandomX(), y = getRandomY()) {
    this.x = x;
    this.y = y;
  }
}
```

Reply:
Related: You could also move the "renderer" part inside the Circle Class:
```
class Circle {
  constructor(x, y) {
     this.x = x;
     this.y = y;
     this.renderer = <CreateCircle />
  }
}

// replaces ents[index] = { core: new Cirlce(), renderer: <CreateCircle /> }
ents[index] = new Circle();

// replaces ents[index].core.update();
ents[index].update();
```